tensorpipe/test/channel/channel_test.h

```diff
@@ -79,45 +80,42 @@ class ChannelTest : public ::testing::TestWithParam<ChannelTestHelper*> {
   }
 
   [[nodiscard]] std::pair<
-      std::future<std::tuple<
-          tensorpipe::Error,
-          tensorpipe::channel::Channel::TDescriptor>>,
+      std::future<
+          std::tuple<tensorpipe::Error, tensorpipe::channel::TDescriptor>>,
       std::future<tensorpipe::Error>>
   sendWithFuture(
-      std::shared_ptr<tensorpipe::channel::Channel> channel,
-      const void* ptr,
-      size_t length) {
+      std::shared_ptr<tensorpipe::channel::Channel<tensorpipe::CpuTensor>>
+          channel,
+      const tensorpipe::CpuTensor& tensor) {
     auto descriptorPromise = std::make_shared<
         std::promise<std::tuple<tensorpipe::Error, std::string>>>();
     auto promise = std::make_shared<std::promise<tensorpipe::Error>>();
     auto descriptorFuture = descriptorPromise->get_future();
     auto future = promise->get_future();
     channel->send(
-        ptr,
-        length,
+        tensor,
         [descriptorPromise{std::move(descriptorPromise)}](
             const tensorpipe::Error& error, std::string descriptor) {
           descriptorPromise->set_value(
               std::make_tuple(error, std::move(descriptor)));
         },
-        [promise{std::move(promise)}](const tensorpipe::Error& error) {
+        [promise{std::move(promise)}, tensor](const tensorpipe::Error& error) {
```

Comment:
Why do we need to capture it?  If it needs to stay alive for the duration of the send, shouldn't we make the channels take it by copy rather than by const-ref? (same below)